### PR TITLE
Fix warnings from Module::CoreList::Utils

### PR DESCRIPTION
**Pull request by atoomic**
This is fixing a warning from Module/CoreList/Utils.pm

```
> ./perl -cw dist/Module-CoreList/lib/Module/CoreList/Utils.pm
Useless use of a constant ("foo") in void context at dist/Module-CoreList/lib/Module/CoreList/Utils.pm line 1665.
```

### CODE REVIEW

**Comment by Leont** [comment]
One could argue that warning is incorrect, it's not entirely a useless use of a constant (just a silly one).

**Comment by atoomic** [comment]
I agree that this is a silly one, but the habit of using a long string to return true at the end of the module is going to have the same issues

`"this is the end of the module, thanks for reading";` and similar should be replaced by `1;` when warnings are enabled

@Leont are you suggesting to fix warnings to authorize such behaviors?

To be honest, I think we should prefer using something like this
`1; # this is my funny final word`

**Comment by hvds** [comment]
I'd argue that it is not even silly - we require only that a module returns a true value to signal success, it is purely convention that has many people returning C<1> for that purpose. The warning is wrong, and should be fixed for this context.

**Comment by atoomic** [comment]
@hvds If we fix the warnings as you suggest, I would then suggest to keep a warning for this context but rephrase it to something more appropriate.

I've the feeling that I've opened a discussion wider than this single pull request, that we should probably move to its own issue.
I've created #17982 to discuss this issue in more generic way

Let's keep the discussion here as: "Is the change from the Pull Request ok or not to merge"

**Comment by hvds** [comment]
Well the merit of this PR is intrinsically connected to the correctness of the warning. My own view is that we should not censor the self-expression of the author (BingoS in this case) to silence a wrong warning.

**Comment by atoomic** [comment]
I would think that core modules should not show bad example of using `foo` for true value.
@bingos any opinion on fixing `s{foo}{1}` here?

**Comment by jkeenan** [comment]
On 7/20/20 12:57 PM, Hugo van der Sanden wrote:
> Well the merit of this PR is intrinsically connected to the correctness 
> of the warning. My own view is that we should not censor the 
> self-expression of the author (BingoS in this case) to silence a wrong 
> warning.

+1.  I always liked those MJD modules that ended:

'cogito ergo sum';

**Comment by Leont** [comment]
> I would think that core modules should not show bad example of using foo for true value.

Why would that be bad?

**Comment by bingos** [comment]
Fine.

I do have literally hundreds of modules on CPAN that do similar, that do not currently warn, and I am not the only one.

**Comment by hvds** [comment]
I'd recommend this PR be put on hold pending resolution of #17921 and/or #17982 .

**Comment by atoomic** [comment]
closing this PR as a more generic choice need to be made first via https://github.com/Perl/perl5/issues/17982
